catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/content/brick/WhenNfcBrickTest.java

```diff
@@ -0,0 +1,488 @@
+		List<String> spinnerValuesResourceIdsContained = Arrays.asList(
+				getResourcesString(R.string.brick_when_nfc_default_all),
+				getResourcesString(R.string.new_nfc_tag),
+				getResourcesString(R.string.test_tag_name_1),
+				getResourcesString(R.string.test_tag_name_2));
+		List<String> spinnerValuesResourceIdsNotContained = Arrays.asList(THIRD_TEST_TAG_ID, FOURTH_TEST_TAG_ID);
+		checkNfcSpinnerContains(nfcBrickPosition, spinnerValuesResourceIdsContained, spinnerValuesResourceIdsNotContained);
+
+		UiHardwareTestUtils.gotoNfcFragment(nfcBrickPosition);
+		NfcTagAdapter adapter = UiHardwareTestUtils.getNfcTagAdapter();
+		adapter.setShowDetails(true);
+
+		UiHardwareTestUtils.fakeNfcTag(THIRD_TEST_TAG_ID, null, null);
+		onView(withText(R.string.default_tag_name)).check(matches(isDisplayed()));
+		onView(withText(NfcHandler.byteArrayToHex(THIRD_TEST_TAG_ID.getBytes())))
+				.check(matches(isDisplayed()));
+		int counterTagsInList = adapter.getCount();
+		assertEquals(3, counterTagsInList);
+		pressBack();
+
+		spinnerValuesResourceIdsContained = Arrays.asList(
+				getResourcesString(R.string.brick_when_nfc_default_all),
+				getResourcesString(R.string.new_nfc_tag),
+				getResourcesString(R.string.test_tag_name_1),
+				getResourcesString(R.string.test_tag_name_2),
+				getResourcesString(R.string.default_tag_name));
+		spinnerValuesResourceIdsNotContained = Arrays.asList(FOURTH_TEST_TAG_ID);
+		checkNfcSpinnerContains(nfcBrickPosition, spinnerValuesResourceIdsContained, spinnerValuesResourceIdsNotContained);
+
+		UiHardwareTestUtils.gotoNfcFragment(nfcBrickPosition);
+		UiHardwareTestUtils.fakeNfcTag(FOURTH_TEST_TAG_ID, null, null);
+		onView(withText(NfcHandler.byteArrayToHex(FOURTH_TEST_TAG_ID.getBytes())))
+				.check(matches(isDisplayed()));
+		counterTagsInList = adapter.getCount();
+		assertEquals(4, counterTagsInList);
+
+		pressBack();
+
+		spinnerValuesResourceIdsContained = Arrays.asList(
+				getResourcesString(R.string.brick_when_nfc_default_all),
+				getResourcesString(R.string.new_nfc_tag),
+				getResourcesString(R.string.test_tag_name_1),
+				getResourcesString(R.string.test_tag_name_2),
+				getResourcesString(R.string.default_tag_name),
+				getResourcesString(R.string.default_tag_name) + "_1");
+		spinnerValuesResourceIdsNotContained = Arrays.asList();
+		checkNfcSpinnerContains(nfcBrickPosition, spinnerValuesResourceIdsContained, spinnerValuesResourceIdsNotContained);
+
+		onScriptList().atPosition(nfcBrickPosition).onChildView(withId(R.id.brick_when_nfc_spinner))
+				.perform(click());
+		onView(withText(R.string.default_tag_name))
+				.check(matches(isDisplayed()));
+	}
+
+	@Test
+	public void testNfcSensorVariable() throws InterpretationException {
+		NdefMessage ndefMessage1 = NfcHandler.createMessage(NFC_NDEF_STRING_1, BrickValues.TNF_MIME_MEDIA);
+		NdefMessage ndefMessage2 = NfcHandler.createMessage(NFC_NDEF_STRING_2, BrickValues.TNF_MIME_MEDIA);
+
+		UiHardwareTestUtils.gotoNfcFragment(nfcBrickPosition);
```

Comment:
either start the activity in the tests instead of setup to then start it with an intent to directly go to the desired fragment, or split the tests in two test classes, the ones where you do checks in the script view, and the ones where you want to check something in the nfcFragment and have setup start the respective fragment / activity